x-pack/plugins/alerting/server/lib/get_rule_notify_when_type.ts

```diff
@@ -10,8 +10,8 @@ import { RuleNotifyWhenType } from '../types';
 export function getRuleNotifyWhenType(
   notifyWhen: RuleNotifyWhenType | null,
   throttle: string | null
-): RuleNotifyWhenType {
+): RuleNotifyWhenType | null {
   // We allow notifyWhen to be null for backwards compatibility. If it is null, determine its
   // value based on whether the throttle is set to a value or null
-  return notifyWhen ? notifyWhen! : throttle ? 'onThrottleInterval' : 'onActiveAlert';
+  return notifyWhen ? notifyWhen! : throttle ? 'onThrottleInterval' : null;
 }
```

Comment:
Do we still need this function?
From 8.7 on, rule level notifyWhen will always be set as null (when creating or updating a rule)... Can't we simply hardcode it to null?

Reply:
We're still allowing the `create` and `update` APIs to be called with these legacy params so I don't think we can remove this function entirely yet.